### ISSUE TITLE
run some tests concurrently where possible; improve package-scripts.js

### ISSUE THREAD

**Issue by boneskull** [mochajs/mocha#3209]
[nps-utils](https://npm.im/nps-utils) makes this pretty easy.

I've cut a lot of time off of running tests locally using concurrent execution.   CI remains to be seen...

**Comment by boneskull** [comment]
[concurrently](https://npmjs.com/concurrently) doesn't seem to allow a limit on *how many processes* to execute at once, which is problematic, since some [just hang](https://travis-ci.org/mochajs/mocha/jobs/330149603).

**Comment by boneskull** [comment]
duplicate of #2839

**Comment by boneskull** [comment]
I'm not seeing any significant speedup on Travis-CI with what I'm doing now, so maybe someone else has a better idea.  To be clear: speeding up CI is the goal.

However, I _am_ seeing a decent speedup in a local environment, which isn't surprising, given the way we partition the tests on Travis-CI.  It's also considerably more difficult to understand what's happening, and *way worse* on Travis-CI, since there are no colors to help.  Here are a couple screenshots:

![image](https://user-images.githubusercontent.com/924465/37843415-0e4d6e0a-2e82-11e8-9902-508c1ce9461d.png)

![image](https://user-images.githubusercontent.com/924465/37843442-23696d02-2e82-11e8-82f6-1e23bba69889.png)

My branch is [here](https://github.com/boneskull/mocha/tree/concurrently) which I'll leave up for awhile, so you can see what I was trying to do.

**Comment by plroebuck** [comment]
You try setting `process.env.FORCE_COLOR = 'true'` first?

**Comment by juergba** [comment]
GH actions is running our CI tests in parallel, or concurrently?
Closing due to inactivity.
